### ISSUE TITLE
Xresources-regolith not applying

### ISSUE THREAD

**Issue by shadowrylander** [regolith-linux/regolith-desktop#95]
Hello! <br>
&nbsp;&nbsp;&nbsp;&nbsp; Before, neither the `regolith i3 config` were applying automatically upon reboot; after commenting out the `rofi` line in the pre-existing `Xresources-regolith` file, both started applying.
&nbsp;&nbsp;&nbsp;&nbsp; However, after the latest update, as before, neither are applying again, despite the `rofi` line still being commented out; neither copying nor linking the `regolith-i3-config` helps, either. Only manually reloading it works, while manually reloading the `Xresources-regolith` file does not. <br>
Any help would be greatly appreciated!

**Comment by kgilmer** [comment]
@shadowrylander if you can wait a few hours, I'm working on publishing a big release that fixes all of this stuff.  I could try and work though the problem with you now but it's all going to change today/tomorrow anyway.  See https://github.com/regolith-linux/regolith-desktop/issues/91 for details.  And if you have and feedback regarding this update, either positive or negative, I'd <3 to hear it.

**Comment by shadowrylander** [comment]
Thank you for the effort! I will update my system as soon as possible; will running an `apt` update + upgrade suffice?

**Comment by kgilmer** [comment]
Yes, you can now `sudo apt update && sudo apt upgrade`` then log back in to your session.  All the content on regolith-linux.org has been updated to the new layout.  See https://www.freelists.org/post/regolith-linux/Config-versioning for more details.

**Comment by shadowrylander** [comment]
I reset my system completely, as I needed to fix a few things in any case, but even after copying over the new i3 config over to `~/.config/i3`, and setting `xrandr --dpi 220:` in my Xresources file and merging it, it still won't apply on reboot. I believe I'm missing a step here...

**Comment by kgilmer** [comment]
Your i3 config file needs to be `~/.config/regolith/i3/config`.  At the moment, Regolith only loads Xresources from `~./Xresources-regolith`, but I'm thinking it should also merge from `~/.Xresources`.  I'll work on that.  The config file path details are covered here: https://regolith-linux.org/troubleshooting.html#where-is-config

**Comment by shadowrylander** [comment]
Ah; it would appear as though an `apt autoremove` removed all necessary packages for rofi, the bar, etc. That means we can never run that command. I'll check my history to restore the packages, but you might want to take a look at that as well. I'll make an update once I've restored them and seen if the dpi settings have taken effect.

**Comment by kgilmer** [comment]
Hmm.  Can you give me an example of a config file that was removed after autoremove?

**Comment by leosunmo** [comment]
Hey @shadowrylander, when you say it removed packages for the bar, do you mean i3xrocks or i3blocks?

Removing Rofi packages seems weird, I updated Regolith yesterday as well and ran autoremove and it only cleaned up deprecated packages. Your config might be referencing old locations or binaries.

**Comment by shadowrylander** [comment]
@khilmer @leosunmo I have a screenshot of the list; is there anything private that can be found in the apt logs? Also, no config files, just packages, and reinstalling them didn't seem to work.

**Comment by leosunmo** [comment]
Yeah I think this is the same issue I had. Remove/back up `~/. Xresources-regolith` and relog.

**Comment by leosunmo** [comment]
Also, if you want to send logs or get more help I can recommend our Slack channel, https://regolith-linux.herokuapp.com/.

**Comment by shadowrylander** [comment]
@leosunmo Er, relog how, again?

**Comment by leosunmo** [comment]
Sorry, a bit unclear, log out and back in again with `gnome-session-quit`.

**Comment by shadowrylander** [comment]
*raises an eyebrow* It worked. But does this mean that I can't use `~/.Xresources-regolith` anymore, or I have to merge it from a different file?

**Comment by shadowrylander** [comment]
Sorry, clicked the wrong button.

**Comment by shadowrylander** [comment]
Also, quick question: what are the permissions of the i3 config file, again? That might be an issue for me.

**Comment by kgilmer** [comment]
I just today updated the init such that Xresources from both `~/.Xresources` and `~/.Xresources-regolith` will be loaded when you session starts.  You may not have gotten that update yet.  Once you do though, you should be able to add things into both files without issue.

Regarding file permissions, the default one in /etc is world readable but only root writable.  Your own copy in `~/.config/regolith/i3/config` should have whatever permissions you prefer.

**Comment by shadowrylander** [comment]
I figured; I don't think I've got it yet.
Also, finally: where exactly should my config file exist? I've currently got it as `~/.config/i3-regolith/config`.

**Comment by shadowrylander** [comment]
In addition, I have a bluetooth issue which might or might not be regolith's fault, where devices connect then disconnect immediately afterwards, as well as the issue of the `apt autoremove`, which I know to remove unnecessary packages; should I create separate issues for them?

**Comment by kgilmer** [comment]
Sure, sounds good.  More likely someone else may find and help out if issues are singled out.

**Comment by shadowrylander** [comment]
Perfect! Also, the config file location. :joy_cat: :

**Comment by kgilmer** [comment]
`~/.config/regolith/i3/config`

**Comment by shadowrylander** [comment]
Just updated the flashback package; tried using `~/.config/regolith/i3/config`, together with `~/.Xresources-regolith`, and my system didn't register the i3 config, leaving me with no keybindings I know of that I could use. Had to drop into a recovery root shell to move them away to `.bak` files. No idea what happened.

**Comment by kgilmer** [comment]
Running this in the terminal will tell you which i3 config file that is currently loaded: ` ps aux | grep -F "i3 -c"`.

**Comment by shadowrylander** [comment]
Oh, no, I can't even get into a terminal by pressing `$mod-Return`. So I can't seem to use neither the i3 config nor the Xresources.

**Comment by kgilmer** [comment]
It's unclear to me what your current situation is @shadowrylander .  A lot has happened in this issue.  Can you summarize your current situation and what you're trying to get done?

**Comment by shadowrylander** [comment]
In a nutshell, I can't seem to use `~/.config/regolith/i3/config` and `~/.Xresources-regolith` without i3 losing all keybindings (for a start), therefore preventing me from using even the terminal without dropping into a root recovery shell. I haven't tried using `~/.Xresources` yet, however.

**Comment by kgilmer** [comment]
The keybindings are defined in the i3 config file.  the Xresources stuff should have no impact on keybindings.  If you use the default Regolith i3 file (`/etc/regolith/i3/config`) (you can get this by renaming or moving `~/.config/regolith/i3/config` and then logging out and back in, you should certainly have keybindings.  Can you send me your `~/.config/regolith/i3/config` file?

**Comment by shadowrylander** [comment]
No problem; and I think that's what I meant, in that I had to fall back to using the default one in `/etc/regolith/i3/config`. Should I just paste the config here? All I did was change the terminal keybinding, then copy it over as `~/.config/regolith/i3/config`.

**Comment by kgilmer** [comment]
You can paste it here or send it to me at kgilmer at gmail.com.  :smile:

**Comment by shadowrylander** [comment]
Huh... And all this time I thought I was using a file with something actually in it... I am terribly sorry about this, but it seems that the file was actually empty all along. Wow am I stupid... It works when it's properly copied. Sorry about that!

**Comment by kgilmer** [comment]
We all make mistakes like that, no worries Jeet.  I'm glad you got things
back up and running!

best
ken

**Comment by shadowrylander** [comment]
Quick question before I close this: is having only `xrandr --dpi 220:` in `~/.Xresources-regolith` a problem? Because it works in `~/.Xresources`.

**Comment by kgilmer** [comment]
You are free to put it in either one.  Both files are loaded at login.  I think `~/.Xresources` is the best choice though as it's a setting not defined by Regolith.

**Comment by shadowrylander** [comment]
Ah; fair enough! Thank you kindly for all your help; I'll go ahead and close this, now!

**Comment by kgilmer** [comment]
Hi there!
 
 Find your next opportunity at an event of your interest
 
 
 
 
 
 10times is the world's largest business event platform to find events,
business conferences, trade shows, global seminars, networking meets and
workshops.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 Events near me
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 Connect with 40m+ event goers world wide.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 300k business events from around the globe.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 Professionals from more than 100 industries.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 Network with event-goers from more than 200 countries & 8000 cities.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 For any queries, reach out to Helpdesk
 
 
 
 
 
 
 Download our official Apps
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 | Manage Email Subscription

Unsubscribe From This List http://links.10times.com/asm/unsubscribe/?user_id=1002061&data=zXVhZhMyF_UmYJLTzxvzwUuxj9LmlKb56VGtyneXc840kOntNGICEEP7ypa1WrKrd0MxoZqS_P_esDvELtflu7EbWYlJK9FlTbEJtI9z39zmJK79gHA_P_WTEm4mk1Zjp6OxVeUHsO-QaBnl7VX-jNWuPq9juElig51lAliCMJbF_DFydICG6zCPhFwm6sd0viGJq7uiZApufC0QVLucZ_kZuCayUHnajbsIKkk1LinaC80ODkx3xuGRqsvU_jduI0M7SJMOXK1QfM-gMoO5QfLgSY1tb0uaUM3rSOk0P9UOHE2QqkkqVwChMUH3lVmh4qb1Sg5KOiCJNOdYJEWorVVyxmLNh7BYjiJQZOXP8zm9gmZ_7lehHy9gblX8V5rYyS_fbpfVPAOiG6A58jatDPn7ZRR6BFsnWghxt-r6g0HhlCvaTyci7XNF6oXo6Q4KAbTtXM48JxqPuFwvcI9aXjCne-o5s5nIx3w8OVNUF6DWyJIkKC4_8C7dtcsFSJnhftarN__AOaN6gnrYwziG0RocWRo9yh7Nkww6iRTZyo0= | Manage Email Preferences http://links.10times.com/asm/?user_id=1002061&data=MSXH1PJ7XdXembPiCInCDnWz-jFbnbd705nPvgdnKDjEmC0JGw8eXRSck3RhqrqE77ah_gXiP4iZSyvxY3WOWunPRtBWENlAymXdrtYFCm8EJjzFRVqF8kvQYrdahOIMKQwzIhVQtoygzrVXusJRTkzhIVkmPOC-zFyny3qeXR8J3wM7HokE8QL1PN5_HJIAfvvDWQsr41U5N6-H4nY_ROH-WMcgdEs-n8DaNO-XbD92azTimaE7WeLtqCvSf_RcueR-xySj8TVMdloCNIWQGOLa2yOPXdIVZU8goEdb0gxSCezVLUsWCk4Nl36uHvijjN6WZNqdbksgrETmCTFEUzZXN9k-f9MH-NJb4IhAyGNS3R-TQJKeSCWx8-4EqCi16N4qOq2ID27WBJ9uSMfg-yp7aG6R3KXFiF4bt-7PlI_9nvplppr6ctLXDoVBkRl1fyfY3vlL8SZVLm5wk_M4HluttRrpqAWf7vRvXiyx5KARyzYoPouOKopDc08w3zDu871TtzRoEq48AGGJqpf-5Q==
